### ISSUE TITLE
Add administrator notes to datasets

### ISSUE THREAD

**Issue by effigies** [OpenNeuroOrg/openneuro#2917]
### What would you like to see added?

It would be useful to have a private field for each dataset where site admins could make notes. Linking GitHub issues and Freshdesk tickets would enable faster identification of previous administrative actions on the dataset. A log could be useful for future admins, if the current administrators move on.

### Do you have any interest in helping implement the feature?

Yes

**Comment by thinknoack** [comment]
@effigies 

just to confirm some assumptions

- field is read/write for admin only.
- should display on dataset page or would you prefer a separate tab
- field should have note, date, author

could this be a admin only comment section/field?

**Comment by nellh** [comment]
I think this would be good to have as part of the dataset event log that's in progress for audit logging and notifications. The implementation I have now has a note associated with each event but also an admin note event which can be created at any time.

The resolver for this returns a timeline of dataset events from this Mongo collection.

```ts
const datasetEventSchema = new Schema<DatasetEventDocument>({
  datasetId: { type: String, required: true },
  timestamp: { type: Date, default: Date.now },
  user: { type: String, required: true },
  event: {
    type: Object,
    required: true,
  },
  success: { type: Boolean, default: false },
  note: { type: String, default: "" },
})
```

The admin note event is just this with the admin flag set. We could support dataset admin notes too? I was only considering it for site admins.

```ts
export type DatasetEventNote = DatasetEventCommon & {
  type: "note"
  // Is this note visible only to site admins?
  admin: boolean
}
```

**Comment by effigies** [comment]
> field is read/write for admin only.

Yes, but I think the way to do this is as being able to add notes to the append-only audit log. I don't think we would want to delete these.

> should display on dataset page or would you prefer a separate tab

Filtering the audit log for notes makes sense to me.

> field should have note, date, author

datetime, but yes.

---

I see @nellh responded while I was writing.

> The admin note event is just this with the admin flag set.

Yes, this is what I was thinking.

> We could support dataset admin notes too? I was only considering it for site admins.

I don't see a use-case for dataset admin-generated audit events. If we want to have site-admin-generated events that the dataset admin can view, they should either be structured actions. If free-text is needed, we should probably be emailing the user.

**Comment by thinknoack** [comment]
@nellh - let's connect on this when/if it makes sense for me to help out with your - audit logging and notifications - work
